pyproject.toml

```diff
@@ -205,7 +205,7 @@ matrix-common = "^1.3.0"
 packaging = ">=16.1"
 # This is the most recent version of Pydantic with available on common distros.
 # We are currently incompatible with >=2.0.0: (https://github.com/matrix-org/synapse/issues/15858)
-pydantic = "^1.7.4"
+pydantic = ">=2.0.0"
```

Comment:
In the context of [our rebuild on Arch Linux](https://archlinux.org/todo/python-pydantic-2/) I have opened a few PRs to fix the remaining projects that I am aware of (also my own).

I realize that for some this might be a bigger change. However, the alternative is using the `v1`, which in part has changed nonetheless and is not the same as using pydantic v1.

From a packager perspective I fear that not upgrading will leave us with a `python-pydantic1` package, that needs to conflict with the `python-pydantic` package (so software that uses either can not be installed with the respective other).

Reply:
As noted elsewhere, introducing a `python-pydantic1` package seems out of the question, as it has to conflict with `python-pydantic` and synapse pulls in `pydantic>2` via twisted -> inflect.

Reply:
So far no other packager of any other distribution has chimed in to raise concerns.
From my perspective I do not see any issues with moving to pydantic v2, as distributions can adapt/update all other pydantic dependents (that I was able to check) appear to be compatible with v2 now (or are about to drop its use).

Reply:
Re. other distributions: 

I am the new pydantic maintainer in Fedora Linux. I am working on updating our development branch (Fedora 40) to pydantic v2, but Fedora 38 and 39 will stay on pydantic v1. It would be nice to add support for pydantic v2 while keeping support for pydantic v1 a little while longer so we can continue to update those releases.

/cc @V02460 who maintains our matrix-synapse package in case they have something to add.